lib/spack/spack/modules.py

```diff
@@ -192,8 +192,6 @@ def parse_config_options(module_generator):
     #####
 
     # Automatic loading loads
-    module_file_actions['hash_length'] = module_configuration.get(
-        'hash_length', 7)
```

Comment:
This, together with other changes in the PR, removes any effect of the `hash_length` keyword in `modules.yaml` and adds the possibility of specifying the hash from the naming scheme. I don't think this is a good idea 

I suppose many people are just fine with the default naming scheme and would like to adjust only the hash length. In the current state of things they can do it adding the line:
```yaml
hash_length: <some-length>
```
to the configuration file. After this PR they will need to write : 
```yaml
'naming_scheme': '${PACKAGE}-${VERSION}-${COMPILERNAME}-${HASH:<some-length>}',
```
which is more boiler-plate for the same amount of information.